tests/test_image_untagger.py

```diff
@@ -0,0 +1,349 @@
+    ]
+    token = "some-token"
+    quay_user = "some-user"
+    quay_password = "some-password"
+    host = "stage.quay.io/"
+    untagger = image_untagger.ImageUntagger(
+        references, token, False, quay_user, quay_password, host
+    )
+
+    lost_digests = untagger.get_lost_digests(
+        ["1", "2"], common_tag_digest_mapping, common_digest_tag_mapping
+    )
+    assert lost_digests == [
+        u"sha256:8a3a33cad0bd33650ba7287a7ec94327d8e47ddf7845c569c80b5c4b20d49d36",
+        u"sha256:2e8f38a0a8d2a450598430fa70c7f0b53aeec991e76c3e29c63add599b4ef7ee",
+        u"sha256:b3f9218fb5839763e62e52ee6567fe331aa1f3c644f9b6f232ff23959257acf9",
+        u"sha256:496fb0ff2057c79254c9dc6ba999608a98219c5c93142569a547277c679e532c",
+    ]
+
+
+def test_untag_images_no_lost_digests(repo_api_data, manifest_list_data, caplog):
+    caplog.set_level(logging.INFO)
+    references = [
+        "stage.quay.io/name/repo1:1",
+    ]
+    token = "some-token"
+    quay_user = "some-user"
+    quay_password = "some-password"
+    host = "stage.quay.io/"
+    untagger = image_untagger.ImageUntagger(
+        references, token, False, quay_user, quay_password, host
+    )
+    with requests_mock.Mocker() as m:
+        m.get(
+            "https://stage.quay.io/api/v1/repository/name/repo1",
+            json=repo_api_data,
+        )
+        m.get(
+            "https://stage.quay.io/v2/name/repo1/manifests/sha256:8a3a33cad0bd33650ba7287a7ec94327d8e47ddf7845c569c80b5c4b20d49d36",
+            json=manifest_list_data,
+            headers={
+                "Content-Type": "application/vnd.docker.distribution.manifest.list.v2+json"
+            },
+        )
+        m.delete("https://stage.quay.io/api/v1/repository/name/repo1/tag/1")
+        lost_images = untagger.untag_images()
+
+        assert lost_images == []
+        assert m.call_count == 4
+
+        expected_logs = [
+            "Gathering tags and digests of repository 'name/repo1'",
+            "Getting manifest list of image stage.quay.io/name/repo1@sha256:8a3a33cad0bd33650ba7287a7ec94327d8e47ddf7845c569c80b5c4b20d49d36",
+            "Getting manifest list of image stage.quay.io/name/repo1@sha256:8a3a33cad0bd33650ba7287a7ec94327d8e47ddf7845c569c80b5c4b20d49d36",
+            "No images will be lost by this untagging operation",
+            "Removing tag '1' from repository 'name/repo1'",
+        ]
+        compare_logs(caplog, expected_logs)
+
+
+def test_untag_images_lost_digests_error(repo_api_data, manifest_list_data, caplog):
+    caplog.set_level(logging.INFO)
+    references = [
+        "stage.quay.io/name/repo1:1",
+        "stage.quay.io/name/repo1:2",
+    ]
+    token = "some-token"
+    quay_user = "some-user"
+    quay_password = "some-password"
+    host = "stage.quay.io/"
+    untagger = image_untagger.ImageUntagger(
+        references, token, False, quay_user, quay_password, host
+    )
+    with requests_mock.Mocker() as m:
+        m.get(
```

Comment:
can you make this request_mocker as fixture?

Reply:
Hmm I don't really see how it would help.

Reply:
mocker setup is repeated over several tests so it can make individual test cases shorter

Reply:
How does one make a fixture with a 'with' statement?

Reply:
like this: https://docs.pytest.org/en/stable/fixture.html#dynamic-scope

Reply:
That doesn't really answer my question, I have to use 'with' statement in the fixture itself: https://requests-mock.readthedocs.io/en/latest/mocker.html